sale_automatic_workflow/models/automatic_workflow_job.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+# © 2011 Akretion Sébastien BEAU <sebastien.beau@akretion.com>
+# © 2013 Camptocamp SA (author: Guewen Baconnier)
+# © 2016 Sodexis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+from contextlib import contextmanager
+from openerp import models, api, fields
+from openerp.tools.safe_eval import safe_eval
+
+_logger = logging.getLogger(__name__)
+
+
+@contextmanager
+def commit(cr):
+    """
+    Commit the cursor after the ``yield``, or rollback it if an
+    exception occurs.
+ is Pending
+    Warning: using this method, the exceptions are logged then discarded.
+    """
+    try:
+        yield
+    except Exception:
+        cr.rollback()
+        _logger.exception('Error during an automatic workflow action.')
+    else:
+        cr.commit()
+
+
+class AutomaticWorkflowJob(models.Model):
+    """ Scheduler that will play automatically the validation of
+    invoices, pickings...  """
+
+    _name = 'automatic.workflow.job'
+
+    @api.model
+    def _validate_sale_orders(self, order_filter):
+        sale_obj = self.env['sale.order']
+        sales = sale_obj.search(order_filter)
+        _logger.debug('Sale Orders to validate: %s', sales.ids)
+        for sale in sales:
+            with commit(self.env.cr):
+                sale.action_confirm()
+
+    @api.model
+    def _create_invoices(self, create_filter):
+        sale_obj = self.env['sale.order']
+        sales = sale_obj.search(create_filter)
+        _logger.debug('Sale Orders to create Invoice: %s', sales.ids)
+        for sale in sales:
+            with commit(self.env.cr):
+                payment = self.env['sale.advance.payment.inv'].create(
+                    {'advance_payment_method': 'all'})
+                payment.with_context(active_ids=sale.ids).create_invoices()
+
+    @api.model
+    def _validate_invoices(self, validate_invoice_filter):
+        invoice_obj = self.env['account.invoice']
+        invoices = invoice_obj.search(validate_invoice_filter)
+        _logger.debug('Invoices to validate: %s', invoices.ids)
+        for invoice in invoices:
+            with commit(self.env.cr):
+                invoice.signal_workflow('invoice_open')
+
+    # TODO:
+    # use account_payment_mode to get the payment information
+    @api.model
+    def _register_payments(self, payment_filter):
```

Comment:
Should it be done in sale_automatic_workflow_payment_mode https://github.com/OCA/sale-workflow/pull/297?